specs/phase0/fork-choice.md

```diff
@@ -459,3 +464,25 @@ def on_attestation(store: Store, attestation: Attestation, is_from_block: bool=F
     # Update latest messages for attesting indices
     update_latest_messages(store, indexed_attestation.attesting_indices, attestation)
 ```
+
+#### `on_attester_slashing`
+
+*Note*: `on_attester_slashing` should be called while syncing and a client MUST maintain the equivocation set of `AttesterSlashing`s from at least the latest finalized checkpoint.
+
+```python
+def on_attester_slashing(store: Store, attester_slashing: AttesterSlashing) -> None:
+    """
+    Run ``on_attester_slashing`` immediately upon receiving a new ``AttesterSlashing``
+    from either within a block or directly on the wire.
+    """
+    attestation_1 = attester_slashing.attestation_1
+    attestation_2 = attester_slashing.attestation_2
+    assert is_slashable_attestation_data(attestation_1.data, attestation_2.data)
+    state = store.block_states[store.justified_checkpoint.root]
```

Comment:
Why do we use the state of the justified block here? It seems strange, as the slashing will have been verified previously against the head state and may fail verification here, or may verify with a different set of slashable indices. In Lighthouse I'm using the head state for this reason, but can change to justified if necessary

Lighthouse comment about this: https://github.com/sigp/lighthouse/blob/94bc85e1f731217c113ff79b427cdeb73b3309cd/consensus/fork_choice/src/fork_choice.rs#L1108-L1116

Lighthouse PR: https://github.com/sigp/lighthouse/pull/3371

cc @adiasg

Reply:
> Why do we use the state of the justified block here?

To maintain uniformity with the fork choice - the [validator set at the justified state](https://github.com/ethereum/consensus-specs/blame/9f643d8dbf678430bc6707f0e5b914d0bf62545c/specs/phase0/fork-choice.md#L177-L183) is used for computing the fork choice. 

> It seems strange, as the slashing will have been verified previously against the head state and may fail verification here, or may verify with a different set of slashable indices.

The intention of `on_attester_slashing` is to censor equivocating validators as soon as an implicating `AttesterSlashing` is received by the node, irrespective of whether those validators are slashable. We are not checking `is_slashable_validator` here (which is quite dependent on the state!), rather only `is_slashable_attestation_data` & `is_valid_indexed_attestation` which do not have much state dependence.

The Lighthouse PR is adding to `store.equivocating_indices` only those validator indices from `attester_slashing` that are slashable in the head state. This should be changed to remove the slashability check.

Reply:
Thank you!

Sorry I don't know what got into me, I got so caught up in re-using `get_slashable_indices` that I didn't realise I shouldn't be filtering already slashed validators at all. Thanks again

Reply:
No worries at all! 😄